client/lib/url/omit-url-params.ts

```diff
@@ -0,0 +1,41 @@
+/**
+ * Internal dependencies
+ */
+import { URL as TypedURL } from 'types';
+import { Falsey } from 'utility-types';
+
+/**
+ * Removes given params from a url.
+ *
+ * @param  url URL to be cleaned
+ * @param  paramsToOmit The collection of params or single param to reject
+ * @return Url less the omitted params.
+ */
+export default function omitUrlParams( url: Falsey, paramsToOmit: string | string[] ): null;
+export default function omitUrlParams( url: TypedURL, paramsToOmit: string | string[] ): TypedURL;
+export default function omitUrlParams(
+	url: TypedURL | Falsey,
+	paramsToOmit: string | string[]
+): TypedURL | null {
+	if ( ! url ) {
+		return null;
+	}
+
+	let toOmit: string[];
+
+	if ( typeof paramsToOmit === 'string' ) {
+		toOmit = [ paramsToOmit as string ];
```

Comment:
Strange that we need this type assertion, I'd expect [`typeof` to work as a type guard](https://www.typescriptlang.org/docs/handbook/advanced-types.html#typeof-type-guards).

Reply:
Looks like it isn't necessary at all. Must have been left behind from when I was trying out different options to handle this. Removed 👍